osutil/syncdir.go

```diff
@@ -63,16 +63,28 @@ var ErrSameState = fmt.Errorf("file state has not changed")
 // exhausted.
 //
 // In all cases, the function returns the first error it has encountered.
-func EnsureDirState(dir, glob string, content map[string]*FileState) (changed, removed []string, err error) {
-	if _, err := filepath.Match(glob, "foo"); err != nil {
-		panic(fmt.Sprintf("EnsureDirState got invalid pattern %q: %s", glob, err))
+func EnsureDirStateMany(dir string, globs []string, content map[string]*FileState) (changed, removed []string, err error) {
+	for _, glob := range globs {
+		if _, err := filepath.Match(glob, "foo"); err != nil {
+			panic(fmt.Sprintf("EnsureDirState got invalid pattern %q: %s", glob, err))
```

Comment:
Why are these using panics? Yes, it's bad, but it's much worse to have a non-functioning system altogether, that can't even revert. Feels like a bad decision we did.

Let's please also prefix the message with "internal error: ".

Reply:
I think you requested those back in the days but I cannot remember for sure.